File: src/TypeScript/TypeScriptLibrary/Library/AliasName.ts
import { IAlias } from "./Interfaces/IAlias";
import { IAliasName } from "./Interfaces/IAliasName";

export class AliasName implements IAliasName
{

    alias !: IAlias;

    name: string = "";

    constructor(alias: IAlias, name: string) {
        this.alias = alias;
        this.name = name;
    }
    getAlias(): IAlias {
        return this.alias;
    }
    getAliasNameValue(): any
    {
        return this.alias.getAliasValue(this.name);
    }


    setAliasNameValue(value: any): void {
    }

    getNameOfAliasName(): string {
        return this.name;
    }
}

File: Java_Generated/Java/untitled/src/general_service/AliasName.java
package general_service;

import general_service.interfaces.IAlias;
import general_service.interfaces.IAliasName;

public class AliasName implements IAliasName {
   IAlias alias;
   String name;

   public AliasName(IAlias alias, String name)
   {
       this.alias = alias;
       this.name = name;
   }

    @Override
    public Object getAliasNameValue() {
        return alias.getAliasValue(name);
    }

    @Override
    public void setAliasNameValue(Object value) {
        alias.setAliasValue(name, value);
    }

    @Override
    public IAlias getAlias() {
        return alias;
    }

    @Override
    public String getNameOfAliasName() {
        return name;
    }
}
